src/extension.ts

```diff
@@ -10,16 +10,24 @@ import * as path from 'path';
 
 import { workspace, ExtensionContext, extensions } from 'vscode';
 import { LanguageClient, LanguageClientOptions, ServerOptions, TransportKind, NotificationType } from 'vscode-languageclient';
-import { URI } from 'vscode-uri';
 import { CUSTOM_SCHEMA_REQUEST, CUSTOM_CONTENT_REQUEST, SchemaExtensionAPI } from './schema-extension-api';
+import { joinPath } from './paths';
 
 export interface ISchemaAssociations {
   [pattern: string]: string[];
 }
 
+export interface ISchemaAssociation {
+  fileMatch: string[];
+  uri: string;
+}
+
 // eslint-disable-next-line @typescript-eslint/no-namespace
 namespace SchemaAssociationNotification {
-  export const type: NotificationType<ISchemaAssociations, unknown> = new NotificationType('json/schemaAssociations');
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  export const type: NotificationType<ISchemaAssociations | ISchemaAssociation[], any> = new NotificationType(
```

Comment:
Why is this an `any` here but the other PR uses `undefined`?

Reply:
The yaml-language-server side notification has any as well:
```
export const type: NotificationType<ISchemaAssociations | SchemaConfiguration[], any> = new NotificationType(
    'json/schemaAssociations'
);
```

Reply:
OK. That is odd, I really I thought I read it as `undefined`.

Reply:
I found it. 
https://github.com/redhat-developer/yaml-language-server/pull/348/files#diff-8a8ae07582c9d433ec8c2e5c4310ff8901e604f4965c5b90a49117ad46c47595R105

Reply:
not that matters, it is not used the same way